streaming/base/dataset.py

```diff
@@ -168,7 +167,12 @@ def __init__(self,
             filename = self._download_file(basename)
         else:
             filename = os.path.join(local, self.split, basename)  # pyright: ignore
-        dist.barrier()
+
+        # Everyone waits for the file to become populated.
+        is_file_exist(filename, TICK, self.download_timeout,
```

Comment:
suggest renaming this function to `wait_for_file_to_exist`

Reply:
Done in https://github.com/mosaicml/streaming/pull/103/commits/bb6649673cbd20eeec770ef30d57668b3ed8afab.